src/Sparrow/Utils/ZstdStream.cs

```diff
@@ -69,15 +70,15 @@ private unsafe int DecompressStep(ReadOnlySpan<byte> buffer)
 
         private unsafe (int OutputPosition, int InputPosition, bool Done) CompressStep(ReadOnlySpan<byte> buffer, ZstdLib.ZSTD_EndDirective directive)
         {
-            lock(this)
+            lock (this)
```

Comment:
do we still need those `lock`s and `if (_compressContext == null)` checks?

Reply:
We still need the `lock` in `Compress` & `Decompress` methods because we may access there from more than one thread and get an `AccessViolationException`, but we can safely remove the `if (_compressContext == null)` checks. Done.

Reply:
Does the `AccessViolationException` is something you are getting when you remove the `lock`? 
I wouldn't expected to have a concurrent usage here...

Reply:
Yes.. it happens for example if we call to `Flush` from `Dispose` method (`ReadWriteCompressStream.cs`) on one thread while we are currently writing to the stream on another.